create-prs-for-vcs-repositories-update/create_prs_for_vcs_repositories_update.py

```diff
@@ -0,0 +1,333 @@
+
+    def __init__(self, token: str):
+        self.g = Github(token)    # cspell: ignore Github
+
+    def url_to_repository_name(self, url:str) -> str:
+        # Get repository name from url
+        match = re.search(GitHubInterface.URL_PATTERN, url)    # cspell: ignore Github
+        assert match is not None, f"URL {url} is invalid"
+        user_name = match.group(1)
+        repo_name = match.group(2)
+
+        return str(f'{user_name}/{repo_name}')
+
+    def get_tags_by_url(self, url: str) -> list[str]:
+        # Extract repository's name from URL
+        repo_name = self.url_to_repository_name(url)
+
+        # Get tags
+        tags: github.PaginatedList.PaginatedList = self.g.get_repo(repo_name).get_tags()
+
+        return [tag.name for tag in tags]
+
+    def create_pull_request(self, repo_name: str, title: str, body: str, head: str, base: str) -> None:
+        # Create a PR from head to base
+        self.g.get_repo(repo_name).create_pull(
+            title=title,
+            body=body,
+            head=head,
+            base=base,
+        )
+
+
+def get_latest_tag(tags: list[str], current_version: str) -> Optional[str]:
+    '''
+    Description:
+        Get the latest tag from the list of tags
+
+    Args:
+        tags (list[str]): a list of tags
+        current_version (str): the current version of the repository
+    '''
+    latest_tag = None
+    for tag in tags:
+        # Exclude parse failed ones such as 'tier4/universe', 'main', ... etc
+        try:
+            version.parse(tag)
+        except (version.InvalidVersion, TypeError):
+            continue
+
+        # OK, it's a valid version
+        if latest_tag is None:
+            latest_tag = tag
+        else:
+            if version.parse(tag) > version.parse(latest_tag):
+                latest_tag = tag
+
+    return latest_tag
+
+
+def create_one_branch(repo: git.Repo, branch_name: str) -> bool:
+
+    # Check if the branch already exists
+    if branch_name in repo.heads:
+        logger.info(f"Branch '{branch_name}' already exists.")
+        return False
+    else:
+        # Create a new branch and checkout
+        repo.create_head(branch_name)
+        logger.info(f"Created a new branch '{branch_name}'")
+        return True
+
+
+def create_version_update_pr(args: argparse.Namespace) -> None:
+
+    # Get GitHub token
+    github_token: str = os.getenv("GITHUB_TOKEN", default=None)
+    if github_token == "None":
+        raise ValueError("Please set GITHUB_TOKEN as an environment variable")
+    github_interface = GitHubInterface(token = github_token)    # cspell: ignore Github
+
+    autoware_repos: AutowareRepos = AutowareRepos(autoware_repos_file_name = args.autoware_repos_file_name)
+
+    # Get the repositories with semantic version tags
+    repository_url_semantic_version_dict: dict[str, str] = autoware_repos.pickup_semver_respositories(semantic_version_pattern = args.semantic_version_pattern)
```

Comment:
typo with respositories

Reply:
Thank you! Now fixed by [this commit](https://github.com/autowarefoundation/autoware-github-actions/pull/317/commits/4c82d6aff1f3d811ef53cb550fb6bcfae9dfee7f).